src/core/mesh/qgsmesh3daveraging.cpp

```diff
@@ -126,6 +128,19 @@ QgsMeshDataBlock QgsMesh3dAveragingMethod::calculate( const QgsMesh3dDataBlock &
         );
       }
     }
+    else if ( singleVerticalIndex != -1 )
```

Comment:
should this be before `if ( !std::isnan( methodLevelTop ) && !std::isnan( methodLevelBottom ) )` ?